pkg/provision/primitives/cache/cache.go

```diff
@@ -248,6 +249,89 @@ func (s *Fs) Exists(id string) (bool, error) {
 	return false, err
 }
 
+// NetworkExists exists checks if a network exists in cache already
+func (s *Fs) NetworkExists(name string, user string) (bool, error) {
+	reservations, err := s.list()
+	if err != nil {
+		return false, err
+	}
+
+	for _, r := range reservations {
+		if r.Type == primitives.NetworkReservation {
+			nr := pkg.NetResource{}
+			if err := json.Unmarshal(r.Data, &nr); err != nil {
+				return false, fmt.Errorf("failed to unmarshal network from reservation: %w", err)
+			}
+
+			// Check if the combination of network name and user is the same
+			if nr.Name == name && r.User == user {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+func (s *Fs) list() ([]*provision.Reservation, error) {
+	s.RLock()
+	defer s.RUnlock()
+
+	infos, err := ioutil.ReadDir(s.root)
+	if err != nil {
+		return nil, err
+	}
+	reservations := make([]*provision.Reservation, len(infos))
+
+	for _, info := range infos {
+		if info.IsDir() {
+			continue
+		}
+
+		r, err := s.get(info.Name())
+		if err != nil {
+			return nil, fmt.Errorf("failed get reservation: %w", err)
+		}
+
+		reservations = append(reservations, r)
```

Comment:
This is wrong here, because you already created a LIST with LEN(infos), so append will just add to the end of the list
for example
```go
l := make([]int, 10)
l = append(l, 42)

assert(l[0] == 0)
assert(l[10] == 42)
```

The right way to do it is
```go
reservations := make([]int, 0, cap)
```
so you create a list with underlying capacity (cap) "len(infos) in  your case" but of length 0 (no elements) 
then you do `append` normally